google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/QueryHintProcessor.java

```diff
@@ -0,0 +1,60 @@
+package com.google.cloud.spanner.hibernate;
+
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Validates and correctly formats Spanner query hints provided through Hibernate.
+ */
+final class QueryHintProcessor {
+
+  /**
+   * Set of all supported Spanner Join hints.
+   */
+  private static final Set<String> JOIN_HINT_KEYS = CollectionUtils.setOf(
+      "FORCE_JOIN_ORDER",
+      "JOIN_METHOD"
+  );
+
+  /**
+   * The pattern that each hint should be written in; i.e. "HINT_KEY=HINT_VALUE".
+   */
+  private static final Pattern HINT_PATTERN = Pattern.compile("([a-zA-Z_]+)=([a-zA-Z_]+)");
```

Comment:
What about whitespace before and after `=`?
Also, is there validation for hints in the client library somewhere?

Reply:
Done; added support for whitespaces before/after '='.

Unfortunately, I didn't find anything useful in the client library even after grepping for 'hint' in google-cloud-spanner.